main.go

```diff
@@ -203,12 +208,14 @@ func (m matrix) String(f format) string {
 	return tableString.String()
 }
 
-func timeHTTPRequest(ctx context.Context, probers []prober.Prober, u *url.URL) *Latency {
+func timeHTTPRequest(ctx context.Context, probers []prober.Prober, u *url.URL, timeout time.Duration) *Latency {
```

Comment:
Why but abstract this and have the timeout already in the given context? That way this function doesn't need to know anything about timeouts

Reply:
the probers are not called async, so each one should get the same amount of time. If we passed a context with timeout, we would wait very long for the first prober and the other probers would never be called

Reply:
Ok well that's a really good reason 👍maybe as a follow up another time we should make the probers all concurrent since they are independent. This would also have the benefit of making the while process faster.

Reply:
I am not sure if we should make all requests at the same time. In many cases only the first one is needed. Ideally, the later probers are never used. So we would generate more (and later ignored) traffic when using all porbers at the same time. Also it is not straight forward which result to use when all of them are successful and result into different times. Or should we cancel the first prober if the second was successful, etc.

Reply:
Ah interesting. Then i misunderstood the probers. I thought this was essentially a list of requests that all needed to be made to satisfy the func

Reply:
Riiight 🙈 it's an interface abstracting _a way_ to time the latency, so as long as the new is successful, it's enough